src/gui/plugins/visualize_lidar/VisualizeLidar.cc

```diff
@@ -453,35 +448,26 @@ void VisualizeLidar::OnScan(const msgs::LaserScan &_msg)
   std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
   if (this->dataPtr->initialized)
   {
-    this->dataPtr->msg = std::move(_msg);
```

Comment:
Also here is quite interesting moment I suppose. `_msg` is const reference. In this case move can't work, it will be copy 
operation https://godbolt.org/z/fjdrKcn9a

Not sure about shenanigans inside `msgs::*` but if it's a fair deep copy it looks like a quite wasteful operation, especially together with `...SetPoints(std::vector<double>(_msg.ranges().begin()....` where we have copy from msg to vector, which will be passed by const reference to SetPoint which also should involve copying data.

I can be wrong, feel free to correct me

Reply:
It's a typical pattern to copy/move messages to avoid blocking the service/topic callback. However, since the mutex is held at the top of the function, this wasn't really accomplishing that. The service would have still been blocked. For the typical pattern to work, we'd also need to be processing the copied message in another thread. You're also right that there woudn't have been a move since it's a const ref.

Overall, the change you've made just to use `_msg` everywhere makes sense.

Reply:
+1, yeah we usually copy the msg and does the processing in another thread to avoid blocking the callback. In this case, it's already processing the msg contents so we shouldn't need to store a copy of the msg in `this->dataPtr->msg` since it's not used anywhere else.

Reply:
Maybe we should add PROFILE macro to callbacks? But maybe it took a fraction of second and can't be too visible on a live top-down view. Not sure